WebRoot/public/login.jsp

```diff
@@ -527,6 +533,17 @@ if (application.getInitParameter("offlineMode") != null) {
                                 <td><label for="password"><fmt:message key="password"/>:</label></td>
                                 <td><input id="password" autocomplete="off" class="zLoginField" name="password" type="password" value="" size="40" maxlength="${domainInfo.webClientMaxInputBufferLength}"/></td>
                                 </tr>
+
+                                <c:if test="${errorCode eq 'account.NEED_CAPTCHA' || errorCode eq 'account.INVALID_CAPTCHA'}">
+                                <tr>
+                                    <td>&nbsp;</td>
+                                    <td>
+                                        <script src='https://www.google.com/recaptcha/api.js'></script>
+                                        <div class="g-recaptcha" data-sitekey="6LdfLE0UAAAAACc89_s1Zw88Hm5_XI5kp-5wimfi"></div>
```

Comment:
same as https://github.com/Zimbra/zm-taglib/pull/6.
secret key must not be hard coded.

Reply:
Note: we'll also need to document the fact that customers need to obtain, then configure ZCS with their own api keys in order to use this functionality.